packages/grafana-runtime/src/utils/queryResponse.ts

```diff
@@ -128,6 +133,25 @@ export function toDataQueryResponse(
   return rsp;
 }
 
+function isCachedResponse(res: FetchResponse<BackendDataSourceResponse | undefined>): boolean {
+  return res?.headers?.get('x-cache') === 'HIT';
+}
+
+function addCacheNotice(frame: DataFrameJSON): DataFrameJSON {
+  const cached: QueryResultMetaNotice = { severity: 'info', text: 'Cached response' };
```

Comment:
**Nit:** probably would rather call this something like `cachedNotice`
```suggestion
  const cachedNotice: QueryResultMetaNotice = { severity: 'info', text: 'Cached response' };
```

Reply:
addressed in [bfd007e](https://github.com/grafana/grafana/pull/45564/commits/bfd007e51fcd7929ffc116b8425c7ab25e48cd10)